adapters/repos/db/lsmkv/bucket_roaring_set_range.go

```diff
@@ -32,3 +43,33 @@ func (b *Bucket) RoaringSetRangeRemove(key uint64, values ...uint64) error {
 
 	return b.active.roaringSetRangeRemove(key, values...)
 }
+
+type ReaderRoaringSetRange interface {
+	Read(ctx context.Context, value uint64, operator filters.Operator) (*sroar.Bitmap, error)
+	Close()
+}
+
+func (b *Bucket) ReaderRoaringSetRange() ReaderRoaringSetRange {
+	MustBeExpectedStrategy(b.strategy, StrategyRoaringSetRange)
+
+	b.flushLock.RLock()
+
+	readers, releaseSegmentGroup := b.disk.newRoaringSetRangeReaders()
+
+	// we have a flush-RLock, so we have the guarantee that the flushing state
+	// will not change for the lifetime of the cursor, thus there can only be two
+	// states: either a flushing memtable currently exists - or it doesn't
+	if b.flushing != nil {
+		readers = append(readers, b.flushing.newRoaringSetRangeReader())
+	}
+	readers = append(readers, b.active.newRoaringSetRangeReader())
+
+	concurrency := 4
```

Comment:
just wondering the reason of the concurrency level

Reply:
that setting indeed could be changed (or explained). will change that to numcpu/2.
with conc=4 I got the best results locally. further increasing concurrency level improved performance only slightly or not at all.